signer/jws.go

```diff
@@ -283,13 +295,13 @@ type jwsProtectedHeader struct {
 	ContentType string `json:"cty"`
 
 	// Lists the headers that implementation MUST understand and process.
-	Critical []string `json:"crit"`
+	Critical []string `json:"crit,omitempty"`
 
 	// The time at which the signature was generated.
 	SigningTime time.Time `json:"io.cncf.notary.signingTime"`
 
 	// The "best by use" time for the artifact, as defined by the signer.
-	Expiry time.Time `json:"io.cncf.notary.expiry,omitempty"`
+	Expiry *time.Time `json:"io.cncf.notary.expiry,omitempty"`
```

Comment:
The zero value of `time.Time` is enough to differentiate between set and unset expiry times, and it is common to not use pointers with this type.

Reply:
Touche, I was completely wrong!